### PR TITLE
[chore] Port all release scripts, CI steps and Makefile from EasyPost-CSharp

### PR DESCRIPTION
**Pull request by nwithan8**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -11,166 +11,241 @@ jobs:
...
+
 
 # .NET Standard notes:
 # - NET Standard 2.0 is compatible with minimum .NET Framework 4.6.1: https://docs.microsoft.com/en-us/dotnet/standard/net-standard?tabs=net-standard-2-0
 # - NET Framework 4.6.1 is EOL after April 26, 2022, due to its security concerns (was affected by the SHA-1 crack): https://dotnet.microsoft.com/en-us/platform/support/policy/dotnet-framework
 # - GitHub's Windows 2022 runner only supports minimum .NET Framework 4.6.2 anyway: https://github.com/actions/virtual-environments/issues/5055#issuecomment-1040757930
 # - .NET Standard is not a framework, but a set of shared APIs between the old .NET Framework and new .NET/.NET Core
-# - `EasyVCR` targets .NET Standard 2.0, meaning it is compatible with specific versions of both .NET and .NET Framework: https://docs.microsoft.com/en-us/dotnet/standard/net-standard?tabs=net-standard-2-0
+# - `EasyPost` targets .NET Standard 2.0, meaning it is compatible with specific versions of both .NET and .NET Framework: https://docs.microsoft.com/en-us/dotnet/standard/net-standard?tabs=net-standard-2-0
```
Comment (Justintime50) [inline .github/workflows/ci.yml:248]: issue: this is bad copy/pasta, you'll want to double check everything and correct as needed

File: scripts/win/setup.bat [hunk 2]
```diff
@@ -10,7 +10,7 @@
 @ECHO OFF
 
 :: .NET Versions we want to install and destination
-SET NetVersions=Current 6.0 5.0 3.1
+SET NetVersions=Current 7.0 6.0 5.0 3.1
```
Comment (Justintime50) [inline scripts/win/setup.bat:13]: question: is this library even setup to support 7.0?

Comment (nwithan8) [reply scripts/win/setup.bat:13]: Yes

File: .github/workflows/ci.yml (hunk 1 above)
Comment (nwithan8) [reply .github/workflows/ci.yml]: Was the only place I missed :)

**Review by Justintime50 — APPROVED** [review]

**Review by jchen293 — APPROVED** [review]

**Pull request merged**